### ISSUE TITLE
OpenSSH doesn't work with proxychains-ng

### ISSUE THREAD

**Issue by NemoAlex** [rofl0r/proxychains-ng#314]
I have 2 VMs  runs same OS: Ubuntu 18.04.3 LTS, one of them has this issue.
ssh client can't work with proxychains, such as:
```
$ proxychains4 ssh git@github.com
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/lib/libproxychains4.so
ssh: Could not resolve hostname github.com: Temporary failure in name resolution
```
Just like it's no proxy at all. But when I switch to root, It works.
The other VM does't has this problem. So weird. Almost same VM, same OS, same user.
Is there anyway to debug?

**Comment by rofl0r** [comment]
- compare proxychains config
- build proxychains with -DDEBUG flag (`CFLAGS=-DDEBUG ./configure ; make`)

**Comment by NemoAlex** [comment]
Proxychains configs are exactly same.

I built proxychains with -DDEBUG flag, nothing different.
```
$ proxychains4 ssh git@github.com
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/local/lib/libproxychains4.so
ssh: Could not resolve hostname github.com: Temporary failure in name resolution
```
Debug works:
```
$ proxychains4 curl qq.com
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/local/lib/libproxychains4.so
DEBUG:init_lib_wrapper called from gcc_init
DEBUG:pid[73640]:at_init
DEBUG:proxy_dns: ON
DEBUG:pid[73640]:wait_data
DEBUG:[play] socks5 127.0.0.1:1080
[proxychains] DLL init: proxychains-ng 4.14-git-8-gb8fa2a7
DEBUG:loaded symbol 'connect' real addr 0x7fe103d8d730  wrapped addr 0x7fe10443e5ef
DEBUG:loaded symbol 'sendto' real addr 0x7fe103d8da10  wrapped addr 0x7fe10443f057
DEBUG:loaded symbol 'gethostbyname' real addr 0x7fe103ac07f0  wrapped addr 0x7fe10443eb66
DEBUG:loaded symbol 'getaddrinfo' real addr 0x7fe103a92bc0  wrapped addr 0x7fe10443ebd2
DEBUG:loaded symbol 'freeaddrinfo' real addr 0x7fe103a93880  wrapped addr 0x7fe10443ec7f
DEBUG:loaded symbol 'gethostbyaddr' real addr 0x7fe103ac00f0  wrapped addr 0x7fe10443ef43
DEBUG:loaded symbol 'getnameinfo' real addr 0x7fe103ac9360  wrapped addr 0x7fe10443a11a
DEBUG:loaded symbol 'close' real addr 0x7fe103d8d410  wrapped addr 0x7fe10443e502
DEBUG:getaddrinfo: qq.com 80
DEBUG:pid[73640]:proxy_getaddrinfo
DEBUG:pid[73640]:proxy_gethostbyname
DEBUG:pid[73640]:wait_data
DEBUG:realloc
DEBUG:creating new entry 0 for ip of qq.com
DEBUG:pid[73640]:wait_data
DEBUG:return hostent space
DEBUG:pid[73640]:mygetservbyname_r
DEBUG:using host getservbyname_r
DEBUG:freeaddrinfo 0x7fe0f0000f30
DEBUG:pid[73640]:proxy_freeaddrinfo
DEBUG:pid[73640]:connect
DEBUG:target: 224.0.0.1
DEBUG:port: 80
DEBUG:pid[73640]:connect_proxy_chain
DEBUG:[play] socks5 127.0.0.1:1080
[proxychains] Strict chain  ...  127.0.0.1:1080 DEBUG:pid[73640]:timed_connect
DEBUG:
connect ret=-1
DEBUG:
poll ret=1
DEBUG:
value=0
DEBUG:pid[73640]:chain_step
DEBUG:pid[73640]:wait_data
DEBUG:pid[73640]:string_from_internal_ip
DEBUG:pid[73640]:index_from_internal_ip
DEBUG:pid[73640]:wait_data
 ...  qq.com:80 DEBUG:pid[73640]:tunnel_to
DEBUG:pid[73640]:wait_data
DEBUG:pid[73640]:string_from_internal_ip
DEBUG:pid[73640]:index_from_internal_ip
DEBUG:pid[73640]:wait_data
DEBUG:host dns qq.com
 ...  OK
<html>
<head><title>302 Found</title></head>
<body bgcolor="white">
<center><h1>302 Found</h1></center>
<hr><center>stgw/1.3.12.4_1.13.5</center>
</body>
</html>
```

**Comment by rofl0r** [comment]
looks as if the .so isn't injected into the process. is it possible the ssh binary is static linked on that machine, or using some sort of wrapper ? (try `file /usr/bin/ssh`)

**Comment by NemoAlex** [comment]
@rofl0r Thanks for your reply .
```
$ which ssh
/usr/bin/ssh
$ file /usr/bin/ssh
/usr/bin/ssh: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/l, for GNU/Linux 3.2.0, BuildID[sha1]=2d66d86a649a5eb5dec8ca70d63752c90deb8177, stripped
```

**Comment by rofl0r** [comment]
> interpreter /lib64/l

that surely sounds bogus to me. does this file exist ?

**Comment by NemoAlex** [comment]
> > interpreter /lib64/l
> 
> that surely sounds bogus to me. does this file exist ?

I just started a new Ubuntu 18.04 on Linode, and confirmed that is by default on 18.04:
```
$ file /usr/bin/ssh
/usr/bin/ssh: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/l, for GNU/Linux 3.2.0, BuildID[sha1]=2d66d86a649a5eb5dec8ca70d63752c90deb8177, stripped
```
The file doesn't exist, I can't explain why.
```
$ ls /lib64/l
ls: cannot access '/lib64/l': No such file or directory
$ ls -l /lib64/ld-linux-x86-64.so.2
lrwxrwxrwx 1 root root 32 Jun  4  2019 /lib64/ld-linux-x86-64.so.2 -> /lib/x86_64-linux-gnu/ld-2.27.so
```

**Comment by NemoAlex** [comment]
The new VM I started works good:
```
$ proxychains4 ssh git@github.com
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/lib/libproxychains4.so
[proxychains] DLL init: proxychains-ng 4.14-git-8-gb8fa2a7
[proxychains] Strict chain  ...  127.0.0.1:1080  ...  github.com:22  ...  OK
The authenticity of host 'github.com (224.0.0.1)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)?
```

**Comment by carlosaguilarmelchor** [comment]
This issue should be closed

**Comment by lbmeng** [comment]
On Ubuntu 22.04, proxychains-ng doesn't work again with ssh:

$ proxychains4 ssh git@github.com
[proxychains] config file found: /etc/proxychains4.conf
[proxychains] preloading /usr/lib/x86_64-linux-gnu/libproxychains.so.4
select2: Bad file descriptor
ssh: Could not resolve hostname github.com: Unknown error

**Comment by rofl0r** [comment]
@lbmeng that is fixed in  latest master see 000000006265577643cdcdefbc94aef903ffaf5e
